diff --git a/libavutil/parseutils.c b/libavutil/parseutils.c
--- a/libavutil/parseutils.c
+++ b/libavutil/parseutils.c
@@ -709,7 +709,12 @@ int av_parse_time(int64_t *timeval, const char *timestr, int duration)
             is_utc = 1;
         }
         if (today) { /* fill in today's date */
-            struct tm dt2 = is_utc ? *gmtime_r(&now, &tmbuf) : *localtime_r(&now, &tmbuf);
+            struct tm dt2;
+            struct tm *dt2_ptr;
+            dt2_ptr = is_utc ? gmtime_r(&now, &tmbuf) : localtime_r(&now, &tmbuf);
+            if (!dt2_ptr)
+                return AVERROR(EINVAL);
+            dt2 = *dt2_ptr;
             dt2.tm_hour = dt.tm_hour;
             dt2.tm_min  = dt.tm_min;
             dt2.tm_sec  = dt.tm_sec;
